pandas/core/indexes/multi.py

```diff
@@ -579,23 +579,22 @@ def _set_names(self, names, level=None, validate=True):
 
         if level is None:
             level = range(self.nlevels)
+            used = set()
         else:
             level = [self._get_level_number(l) for l in level]
+            used = set(self.levels[l].name
+                       for l in set(range(self.nlevels)) - set(level))
 
         # set the name
         for l, name in zip(level, names):
+            if name is not None and name in used:
+                raise ValueError("Duplicated level name: {}.".format(name))
```

Comment:
can you enumerate on this and report the level number where the error is as well
e.g. ``level 2, duplicate name 'foo'`` or somesuch